src/main/java/core/basesyntax/strategy/handler/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private final FruitDao fruitDao;
+
+    public PurchaseOperationHandler(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public void process(FruitTransaction transaction) {
+        Integer currentQuantity = fruitDao.get(transaction.getFruit());
+        int purchaseQuantity = transaction.getQuantity();
+        if (purchaseQuantity <= 0 && purchaseQuantity > currentQuantity) {
```

Comment:
The condition `purchaseQuantity <= 0 && purchaseQuantity > currentQuantity` will always evaluate to false because the purchase quantity cannot be both less than or equal to 0 and greater than the current quantity at the same time. You should separate these conditions into two different checks: one for validating that the purchase quantity is positive and another for ensuring that there is enough stock to complete the purchase.